src/wazuh_qa_framework/wazuh_components/api/wazuh_api.py

```
@@ -113,12 +113,468 @@ def list_agents(self, response):
         """
         return response.data
 
+    @WazuhAPIRequest(method='GET', endpoint='/agents')
+    def get_agent_id(self, response, host_ip):
+        """Get agent ID.
+
+        Returns:
+            str: Agent ID.
+        """
+        for agent in response.data['affected_items']:
+            if host_ip in agent['ip']:
+                return agent['id']
```

Should we raise an exception in case the agent is not registered?

---

Done in a0c5f44